### PR TITLE
Enhancing topoUp plugin

### PR DESCRIPTION
**Pull request by karmoham**

### CODE REVIEW

**Comment by karmoham** [comment]
[failed_log.txt](https://github.com/CiscoTestAutomation/pyats.contrib/files/5318788/failed_log.txt)
[Successfull_connection.txt](https://github.com/CiscoTestAutomation/pyats.contrib/files/5318789/Successfull_connection.txt)

File: src/pyats/contrib/plugins/topoup_plugin/topoup.py [hunk 1]
```diff
@@ -86,9 +87,24 @@ def pre_job(self, task):
...
+
+        failed = False
+
+        for res, dev, count in pcall_output:
+            if res is False:
+                failed = True
+                log.warning("Device '{device}' connectivity check failed after '{count}' trial(s)".format(
+                    device=dev, count=count))
+            else:
+                log.info("Device '{device}' connectivity passed after '{count}' trial(s)".format(
```
Comment (jeaubin) [inline src/pyats/contrib/plugins/topoup_plugin/topoup.py:101]: can we have them grouped

the passed one first 
then the failed one

File: src/pyats/contrib/plugins/topoup_plugin/topoup.py [hunk 2]
```diff
@@ -86,9 +87,24 @@ def pre_job(self, task):
...
+                    device=dev, count=count))
+
+        if failed:
             # Terminate testscript
             raise Exception ("Not all the testbed devices are up and ready")
+            # log.error("Not all the testbed devices are up and ready")
```
Comment (jeaubin) [inline src/pyats/contrib/plugins/topoup_plugin/topoup.py:107]: remove comment

File: src/pyats/contrib/plugins/topoup_plugin/topoup.py (hunk 1 above)
Comment (karmoham) [reply src/pyats/contrib/plugins/topoup_plugin/topoup.py]: image

image

**Review by jeaubin — APPROVED** [review]

**Pull request merged**